### PR TITLE
Range-ify std.path.stripExtension()

### PR DESCRIPTION
**Pull request by WalterBright**
http://dlang.org/phobos/std_path.html#.stripExtension

And removed some redundant unit tests.

### CODE REVIEW

File: std/path.d [hunk 1]
```diff
@@ -757,21 +757,60 @@ unittest
...
 private ptrdiff_t extSeparatorPos(R)(const R path)
-    if (isRandomAccessRange!R && hasSlicing!R && isSomeChar!(ElementType!R) ||
+    if (isRandomAccessRange!R && hasLength!R && isSomeChar!(ElementType!R) ||
         isNarrowString!R)
 {
-    auto i = (cast(ptrdiff_t) path.length) - 1;
-    while (i >= 0 && !isSeparator(path[i]))
+    for (auto i = cast(ptrdiff_t)path.length - 1;
+         i >= 0 && !isSeparator(path[i]);
+         --i)
```
Comment (andralex) [inline std/path.d]: The C "downto" idiom works nicely here:

    for (auto i = path.length; i-- > 0 && !isSeparator(path[i]); )

**Comment by andralex** [comment]
lgtm modulo the optional nit

**Comment by andralex** [comment]
berry nice

**Comment by andralex** [comment]
Auto-merge toggled on

**Comment by WalterBright** [comment]
Oh joy, FreeBSD heisenbugs. https://issues.dlang.org/show_bug.cgi?id=14647

**Pull request merged**